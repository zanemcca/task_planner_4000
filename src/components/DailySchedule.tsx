import React from 'react';
import {
  Row, Card
} from 'antd';
import {
  map
} from 'lodash';
import moment from 'moment';

import '../styles/DailySchedule.css';
import Task, { ITask} from './Task';

export interface IDailySchedule {
  date: moment.Moment
  tasks: ITask[]
}

export type IDailyScheduleProps = IDailySchedule

const DailySchedule = (props: IDailyScheduleProps) => {
  return (
    <Card className="DailySchedule">
      <Row className="DailySchedule-day">{props.date.format('dddd')}</Row>
      <Row className="DailySchedule-date">{props.date.format('MMMM D')}</Row>
      {map(props.tasks, task => (
        <Row key={task.description}>
          <Task {...task}/>
        </Row>
      ))}
    </Card>
  );
}

export default DailySchedule;
