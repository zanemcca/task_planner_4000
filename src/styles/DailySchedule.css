.DailySchedule {
}

.DailySchedule-day {
  font-family: Avenir;
  font-size: 24px;
  line-height: 33px;
  color: #333333;
}

.DailySchedule-date {
  font-family: Avenir;
  font-size: 14px;
  line-height: 17px;
  color: #787878;
}
