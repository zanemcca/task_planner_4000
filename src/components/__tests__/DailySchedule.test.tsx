import renderer from 'react-test-renderer';

import DailySchedule, { IDailyScheduleProps } from '../DailySchedule';
import moment from 'moment';
import React from 'react';

it('should render multiple tasks in a nice way', () => {
  const props: IDailyScheduleProps = {
    date: moment('2019-10-10'),
    tasks: [{
      description: 'Aliquam vel massa accumsan, accumsan ex eu, posuere nunc.',
      channel: {
        color: 'red',
        title: 'redchannel'
      },
      time: moment.duration('00:30:00'),
    }, {
      description: 'Nam malesuada massa odio.',
      channel: {
        color: 'blue',
        title: 'bluechannel'
      },
    }]
  }

  const component = renderer.create( <DailySchedule {...props}/>)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
