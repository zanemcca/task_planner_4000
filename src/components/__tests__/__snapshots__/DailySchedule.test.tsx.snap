// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render multiple tasks in a nice way 1`] = `
<div
  className="ant-card DailySchedule ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-row DailySchedule-day"
      style={Object {}}
    >
      Thursday
    </div>
    <div
      className="ant-row DailySchedule-date"
      style={Object {}}
    >
      October 10
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-card Task ant-card-bordered"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-row Task-header"
            style={Object {}}
          >
            <div
              className="Task-description"
            >
              Aliquam vel massa accumsan, accumsan ex eu, posuere nunc.
            </div>
            <div
              className="Task-time"
            >
              30 minutes
            </div>
          </div>
          <div
            className="ant-row Task-footer"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-3 Task-complete"
              style={Object {}}
            >
              <i
                aria-label="icon: check-circle"
                className="anticon anticon-check-circle"
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                </svg>
              </i>
            </div>
            <div
              className="ant-col ant-col-21 Task-channel"
              style={Object {}}
            >
              <div>
                <span
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  # 
                </span>
                <span>
                  redchannel
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-card Task ant-card-bordered"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-row Task-header"
            style={Object {}}
          >
            <div
              className="Task-description"
            >
              Nam malesuada massa odio.
            </div>
          </div>
          <div
            className="ant-row Task-footer"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-3 Task-complete"
              style={Object {}}
            >
              <i
                aria-label="icon: check-circle"
                className="anticon anticon-check-circle"
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                </svg>
              </i>
            </div>
            <div
              className="ant-col ant-col-21 Task-channel"
              style={Object {}}
            >
              <div>
                <span
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                >
                  # 
                </span>
                <span>
                  bluechannel
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
